lib/stdlib/src/maps.erl

```diff
@@ -194,6 +194,25 @@ merge_with_1({K, V2, Iterator}, Map1, Map2, Combiner) ->
 merge_with_1(none, Result, _, _) ->
     Result.
 
+-spec merge_deep(Map1,Map2) -> Map3 when
+    Map1 :: map(),
+    Map2 :: map(),
+    Map3 :: map().
+
+merge_deep(Map1, Map2) when is_map(Map1), is_map(Map2) ->
+    Combiner = fun (_Key1, Value1, Value2) ->
+        case is_map(Value1) andalso is_map(Value2) of
+            true ->
+                merge_deep(Value1, Value2);
+            false ->
+                Value2
+        end
+    end,
```

Comment:
```suggestion
    Combiner = fun
        (_Key1, Value1, Value2) when is_map(Value1) andalso is_map(Value2) ->
                merge_deep(Value1, Value2);
         (_Key1, _Value1, Value2) ->
                Value2
        end
    end,
```

Reply:
Don't use `andalso` in guards unless really necessary (nested expressions with parentheses). So in this case
```erlang
  Combiner = fun
        (_Key1, Value1, Value2) when is_map(Value1),  is_map(Value2) ->
```